### PR TITLE
feat!: support non-trio examples in the registry

### PR DESCRIPTION
**Pull request by samestep**
# Description

Following up on #1393 to prepare for #1391, this PR creates a `@penrose/solids` package to hold [SolidJS](https://www.solidjs.com/) examples using [Babel](https://babeljs.io/) to compile the JSX for both Node and browser, adds that as a dependency of `@penrose/examples`, and moves the registry runner code from `@penrose/roger` to a Vitest suite in `@penrose/examples` that can now run arbitrary non-trio examples right alongside trio examples.

# Implementation strategy and design decisions

Previously `@penrose/examples` was a dev dependency of `@penrose/core`, used for a few tests; this PR would have created a circular dependency, so I just removed those core tests because they're basically subsumed by the registry anyways.

# Examples with steps to reproduce them

Now running all the registry examples locally is very easy:

```sh
yarn registry
```

# Checklist

- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have reviewed any generated registry diagram changes

### CODE REVIEW

**Review by wodeni — CHANGES_REQUESTED** [review]
Looks great! Everything seems good to me. Can you edit the README for `roger` to remove the `batch` command and change the example in the README from `batch` to `trio`?

**Review by wodeni — CHANGES_REQUESTED** [review]
Looks great! Everything seems good to me. Can you edit the README for `roger` to remove the `batch` command and change the example in the README from `batch` to `trio`?

File: packages/examples/src/registry.test.ts [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+        return fileReq.text();
+      } catch (e) {
+        console.error(`Failed to resolve path: ${e}`);
+        return undefined;
+      }
+    }
+    return await resolver(filePath);
+  };
+
+  // TODO: change "roger" to "registry"
```
[the remark this answers is not in the document]
Comment (samestep) [reply packages/examples/src/registry.test.ts:98]: Yeah, do you want me to do it in this PR? I was mostly just procrastinating on the big noisy diff.

**Comment by samestep** [comment]
> Looks great! Everything seems good to me. Can you edit the README for `roger` to remove the `batch` command and change the example in the README from `batch` to `trio`?

Done, thanks!

File: packages/examples/src/registry.test.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (wodeni) [reply packages/examples/src/registry.test.ts:98]: Sure. I've seen the diffs (just that one file) and it seems good. I'll go over the diagrams too after you make that change

**Pull request merged**